src/opendr/perception/panoptic_segmentation/efficient_ps/efficient_ps_learner.py

```diff
@@ -0,0 +1,521 @@
+            logging_path: str=str(Path(__file__).parent / 'logging'),
+            silent: bool=False,
+            verbose: Optional[bool]=None
+            ):
+        """
+        This method is used for training the algorithm on a train dataset and validating on a separate dataset.
+
+        :param dataset: Object that holds the training dataset
+        :type dataset: Dataset class type
+        :param val_dataset: Object that holds the validation dataset
+        :type val_dataset: Dataset class type, optional
+        :param logging_path: Path to store the logging files, e.g., training progress and tensorboard logs
+        :type logging_path: str
+        :param silent: if True, disables printing training progress reports to STDOUT. The evaluation will still be shown.
+        :type silent: bool
+        """
+        if verbose is not None:
+            warnings.warn('The verbose parameter is not supported and will be ignored.')
+
+        self._cfg.work_dir = logging_path
+
+        dataset.pipeline = self._cfg.train_pipeline
+        dataloaders = [build_dataloader(
+            dataset.get_mmdet_dataset(),
+            self.batch_size,
+            self.num_workers,
+            self._cfg.gpus,
+            dist=False,
+            seed=self._cfg.seed
+        )]
+
+        # Put model on GPUs
+        self.model = MMDataParallel(self.model, device_ids=range(self._cfg.gpus)).cuda()
+
+        optimizer = build_optimizer(self.model, self._cfg.optimizer)
+        if silent:
+            logger = get_root_logger(log_level=logging.WARN)
+        else:
+            logger = get_root_logger(log_level=logging.INFO)
+
+        # Record some important information such as environment info and seed
+        env_info_dict = collect_env()
+        env_info = '\n'.join([('{}: {}'.format(k, v)) for k, v in env_info_dict.items()])
+        meta = {'env_info': env_info, 'seed': self._cfg.seed}
+
+        runner = Runner(
+            self.model,
+            batch_processor,
+            optimizer,
+            self._cfg.work_dir,
+            logger=logger,
+            meta=meta
+        )
+
+        timestamp = time.strftime('%Y%m%d_%H%M%S', time.localtime())
+        runner.timestamp = timestamp
+        runner.register_training_hooks(self._cfg.lr_config, self._cfg.optimizer_config,
+                                       self._cfg.checkpoint_config, self._cfg.log_config)
+
+        if val_dataset is not None:
+            val_dataset.pipeline = self._cfg.test_pipeline
+            val_dataloader = build_dataloader(
+                val_dataset.get_mmdet_dataset(test_mode=True),
+                imgs_per_gpu=1,
+                workers_per_gpu=self.num_workers,
+                dist=False,
+                shuffle=False
+            )
+            runner.register_hook(EvalHook(val_dataloader, interval=1, metric=['panoptic']))
+
+        runner.run(dataloaders, self._cfg.workflow, self.iters)
+        self._is_model_trained = True
```

Comment:
I think *fit()* is also expected to return a dictionary with the training statistics.

Reply:
I addressed this by returning the training (losses) and validation (metrics) statistics. The description in D2.1 does not directly translate to my use case.